### PR TITLE
Add model deviation management for report class

### PR DESCRIPTION
**Pull request by Angel-Jia**
1. Add model deviation management for report.
2. Save model deviations in report.

### CODE REVIEW

File: dpgen2/exploration/deviation/deviation_std.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    The model deviation can be List[None], where len(List[None]) is
+    the number of trajectory files.
+
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._data = defaultdict(list)
+
+    def _add(self, name: str, deviation: np.ndarray) -> None:
```
Comment (wanghan-iapcm) [inline dpgen2/exploration/deviation/deviation_std.py:35]: Adding frames one by one may be dangerous. The user is allowed to add to different items different numbers of frames, then the frames will never be aligned. 

I would suggest adding all the frames once, i.e. deviation is List[np.ndarray] or a np.ndarray of dim==2

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]

File: dpgen2/exploration/deviation/deviation_std.py (hunk 1 above)
Comment (Angel-Jia) [reply dpgen2/exploration/deviation/deviation_std.py:35]: @wanghan-iapcm How about adding a `check_data` function?

Comment (Angel-Jia) [reply dpgen2/exploration/deviation/deviation_std.py:35]: @wanghan-iapcm How about adding a `check_data` function like this:

Comment (Angel-Jia) [reply dpgen2/exploration/deviation/deviation_std.py:35]:
```
@abstractmethod
def check_data(self):
    r""" Check if data is valid
    """
    pass
```
one must call `check_data` function before get data from DeviManager.

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]
I agree with your implementation. 

We should also check the length of the arrays corresponding to the same trajectory has the same number of frames.

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**